src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/_validators.py

```diff
@@ -4,9 +4,19 @@
 # --------------------------------------------------------------------------------------------
 
 from datetime import datetime
+from decimal import Decimal
 from azure.cli.core.util import CLIError
 
 
+def get_decimal_type():
+    def decimal_type(string):
+        try:
+            return Decimal(string)
+        except ValueError:
+            raise ValueError("the value passed cannot be converted to decimal")
+    return decimal_type
```

Comment:
Is there a reason you don't simple have this argument registered with `type=Decimal`?

Reply:
We have some decimal fields. Conversion from and to string makes data consistent and comparable